### PR TITLE
Throttle correction

### PR DESCRIPTION
**Pull request by chandumlg**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -343,7 +343,7 @@ func (worker *workerT) enhanceResponse(rawMsg []byte, key, val string) (resp []b
 	}()
 
 	var existingResponse string
-	if len(rawMsg) != 0 {
+	if len(rawMsg) > 2 {
```
[the remark this answers is not in the document]
Comment (SumanthPuram) [reply router/router.go:346]: Using sjson

File: router/router.go [hunk 2]
```diff
@@ -418,18 +418,23 @@ func (worker *workerT) handleWorkerDestinationJobs() {
...
 		var attemptedToSendTheJob bool
-		payload := []byte(`{}`)
+		payload := []byte{}
 		if destinationJob.StatusCode == 200 || destinationJob.StatusCode == 0 {
 			payload = destinationJob.Message
 			if worker.canSendJobToDestination(prevRespStatusCode, failedUserIDsMap, destinationJob) {
 				diagnosisStartTime := time.Now()
+				sourceID := destinationJob.JobMetadataArray[0].SourceID
+				destinationID := destinationJob.JobMetadataArray[0].DestinationID
+				userID := ""
```
Comment (SumanthPuram) [inline router/router.go:428]: Batched jobs and router transformed jobs can have more than 1 metadatas. So, don't increment userID.

File: router/router.go [hunk 3]
```diff
@@ -576,7 +588,9 @@ func (worker *workerT) postStatusOnResponseQ(respStatusCode int, respBody string
 	if isSuccessStatus(respStatusCode) {
 		atomic.AddUint64(&worker.rt.successCount, 1)
 		status.JobState = jobsdb.Succeeded.State
-		status.ErrorResponse = worker.enhanceResponse(status.ErrorResponse, "reason", respBody)
+		if respBody != "" {
```
Comment (SumanthPuram) [inline router/router.go:591]: Let's move this above.

File: router/router.go [hunk 4]
```diff
@@ -509,7 +514,14 @@ func (worker *workerT) handleWorkerDestinationJobs() {
 
 			//Not saving payload to DB if transformAt is not "router"
 			if destinationJobMetadata.TransformAt != "router" {
-				payload = []byte(`Same as payload in router table. Use job_id for look up.`)
+				payload = []byte{}
+			}
+
+			var errorResponse []byte
```
Comment (SumanthPuram) [inline router/router.go:520]: Move into postStatusOnResponseQ(), save payload for both batched and dest transformed jobs

File: router/router.go (hunk 2 above)
Comment (chandumlg) [reply router/router.go:428]: added comment.

File: router/throttler/throttler.go [hunk 5]
```diff
@@ -114,18 +115,19 @@ func (throttler *HandleT) LimitReached(destID string, userID string) bool {
...
-	if !limitReached {
-		if throttler.destLimiter.enabled {
-			throttler.destLimiter.ratelimiter.Inc(destKey)
-		}
-		if throttler.userLimiter.enabled {
-			throttler.userLimiter.ratelimiter.Inc(userKey)
-		}
-	}
+func (throttler *HandleT) Inc(destID string, userID string) {
+	destKey := fmt.Sprintf(`%s_%s`, throttler.destinationName, destID)
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/throttler/throttler.go:122]: done

File: router/router.go (hunk 3 above)
Comment (chandumlg) [reply router/router.go:591]: done

File: router/router.go (hunk 4 above)
Comment (chandumlg) [reply router/router.go:520]: done

File: router/router.go [hunk 6]
```diff
@@ -418,18 +418,23 @@ func (worker *workerT) handleWorkerDestinationJobs() {
...
-		payload := []byte(`{}`)
+		payload := []byte{}
 		if destinationJob.StatusCode == 200 || destinationJob.StatusCode == 0 {
 			payload = destinationJob.Message
 			if worker.canSendJobToDestination(prevRespStatusCode, failedUserIDsMap, destinationJob) {
 				diagnosisStartTime := time.Now()
+				sourceID := destinationJob.JobMetadataArray[0].SourceID
+				destinationID := destinationJob.JobMetadataArray[0].DestinationID
+				userID := ""
+				if len(destinationJob.JobMetadataArray) == 1 {
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go:429]: Batch or router transform

File: router/router.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go:346]: done

**Review by SumanthPuram — APPROVED** [review]

**Pull request merged**